pkg/component/kubeapiserver/kube_apiserver.go

```diff
@@ -575,3 +578,45 @@ func getLabels() map[string]string {
 		v1beta1constants.LabelRole: v1beta1constants.LabelAPIServer,
 	}
 }
+
+// ComputeKubeAPIServerServiceAccountConfig computes the [ServiceAccountConfig]
+// needed to configure a kube-apiserver.
+func ComputeKubeAPIServerServiceAccountConfig(
+	config *gardencorev1beta1.ServiceAccountConfig,
+	externalHostname string,
+	serviceAccountKeyRotationPhase gardencorev1beta1.CredentialsRotationPhase,
+) ServiceAccountConfig {
+	var (
+		defaultIssuer = "https://" + externalHostname
+		out           = ServiceAccountConfig{
+			Issuer:        defaultIssuer,
+			RotationPhase: serviceAccountKeyRotationPhase,
+		}
+	)
+
+	if config == nil {
+		return out
+	}
+
+	out.ExtendTokenExpiration = config.ExtendTokenExpiration
+	out.MaxTokenExpiration = config.MaxTokenExpiration
+
+	if config.Issuer != nil {
+		out.Issuer = *config.Issuer
+	}
+	out.AcceptedIssuers = config.AcceptedIssuers
+	if out.Issuer != defaultIssuer && !slices.Contains(out.AcceptedIssuers, defaultIssuer) {
+		out.AcceptedIssuers = append(out.AcceptedIssuers, defaultIssuer)
+	}
```

Comment:
This behaviour was introduced with https://github.com/gardener/gardener/pull/5888. I preserved the original behaviour. Also there is no guarantee when the tokens with "old" issuer are all refreshed.

Reply:
🤔 , ok default issuer was set to be always accepted to keep tokens acceptable when a custom issuer is set. However, there is nothing in this direction when the custom issuer is changed, except the shoot owner is responsible to set it for some time as `acceptedIssuer`. 

What is the strategy now for setting a managed issuer? Shoot owner is responsible to properly configure accepted issuer  when annotates the shoot with `authentication.gardener.cloud/issuer=managed`?

Reply:
> What is the strategy now for setting a managed issuer? Shoot owner is responsible to properly configure accepted issuer when annotates the shoot with authentication.gardener.cloud/issuer=managed?

Yes, if you have a custom issuer then you should put it in the `acceptedIssuers` once you migrate to a managed one. This will be documented later on

Reply:
Please adapt https://github.com/gardener/gardener/blob/master/docs/usage/shoot_serviceaccounts.md accordingly.

Reply:
What is the point of putting something in the documentation before it is implemented? I am afraid that someone will read it and decide that this feature is already supported. I would rather put this in a consecutive PR, WDYT?